content/updates/terraform-null-label-advanced.md

```diff
@@ -0,0 +1,343 @@
+---
+visible: true
+draft: false
+title: "terraform-null-label: Advanced Usage"
+author: Matt Gowie
+date: 2024-03-07
+description: "A post highlighting some advanced usage of the terraform-null-label module showing root/child module relationship and implementation of a naming + tagging framework with `context.tf`"
+image: /img/updates/terraform-null-label-part2.png
+---
+
+Following up from our last article on terraform-null-label, (if you haven’t read that already, [be sure to here](https://masterpoint.io/updates/terraform-null-label/)) we want to talk about some advanced usage scenarios. Terraform-null-label is great, but it can still be cumbersome to implement when you need to create lots of variables that implement the label interface. In that case, why not borrow a little bit from object-oriented (OO) programming with [the concept of a "mixin"](https://en.wikipedia.org/wiki/Mixin)? Terraform-null-label has this capability through the drop-in `context.tf` file. It provides a great framework to implement naming and tagging, especially in scenarios where there’s lots of nesting and module dependencies. In this follow-up, we’ll explore mixins, and provide some code examples to help understand how to get the most out of terraform-null-label.
```

Comment:
```suggestion
Following up from our last article on terraform-null-label, (if you haven’t read that already, [be sure to here](https://masterpoint.io/updates/terraform-null-label/)) we want to talk about some advanced usage scenarios. Terraform-null-label is great, but it can still be cumbersome to implement when you need to create lots of variables that implement the label interface. In that case, why not borrow a little bit from object-oriented (OO) programming with [the concept of a "mixin"](https://en.wikipedia.org/wiki/Mixin)? Terraform-null-label has this capability through the drop-in `context.tf` file. It provides a great framework to implement naming and tagging, especially in scenarios where there’s lots of nesting and module dependencies. In this follow-up, we’ll explore mixins, and provide some code examples to help understand how to get the most out of terraform-null-label.
```
```suggestion
Following up from our last article on the `terraform-null-label` module, (if you haven’t read that already, [be sure to here](https://masterpoint.io/updates/terraform-null-label/)) we want to talk about some advanced usage scenarios. Terraform-null-label is great, but it can still be cumbersome to implement when you need to create lots of variables that implement the label interface. In that case, why not borrow a little bit from object-oriented (OO) programming with [the concept of a "mixin"](https://en.wikipedia.org/wiki/Mixin)? Terraform-null-label has this capability through the drop-in `context.tf` file. It provides a great framework to implement naming and tagging, especially in scenarios where there’s lots of nesting and module dependencies. In this follow-up, we’ll explore mixins, and provide some code examples to help understand how to get the most out of terraform-null-label.
```